app/models/course/condition/assessment.rb

```diff
@@ -0,0 +1,41 @@
+class Course::Condition::Assessment < ActiveRecord::Base
+  include ActiveSupport::NumberHelper
+
+  acts_as_condition
+  belongs_to :assessment, class_name: Course::Assessment.name, inverse_of: false
+
+  default_scope { includes(:assessment) }
+
+  def title
+    if minimum_grade_percentage
+      minimum_grade_percentage_display = number_to_percentage(minimum_grade_percentage,
+                                                              precision: 2,
+                                                              strip_insignificant_zeros: true)
+      self.class.human_attribute_name('title.title',
+                                      assessment_title: assessment.title,
+                                      minimum_grade_percentage: minimum_grade_percentage_display)
+    else
+      assessment.title
+    end
+  end
+
+  def satisfied_by?(course_user)
+    graded_submission = graded_submission_by_user(course_user.user)
+
+    if minimum_grade_percentage
+      graded_submission.present? && grade_percentage(graded_submission) >= minimum_grade_percentage
+    else
+      graded_submission.present?
+    end
+  end
+
+  private
+
+  def graded_submission_by_user(user)
+    assessment.submissions.by_user(user).find_by_workflow_state(:graded)
```

Comment:
What if the user has multiple submissions for that one assessment? Possible for trainings.